doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+resource        : Object which generated the event {string}
+severity        : Severity of the event {string}
+acknowledged    : Indicates when alarm has been acknowledged/unacknowledged {uint64}
+
+127.0.0.1:6379[6]> hgetall "ALARM|2"
+ 1) "type-id"
+ 2) "TEMPERATURE_EXCEEDED"
+ 3) "text"
+ 4) "temperatureCrossedThreshold: Current temperature for sensor/2 is 76 degrees"
+ 5) "action"
+ 6) "RAISE"
+ 7) "resource"
+ 8) "sensor/2"
+ 9) "time-created"
+10) "1621460371062299951"
+11) "severity"
+12) "CRITICAL"
+13) "id"
+14) "2"
+15) "acknowledged"
+16) "false"
+127.0.0.1:6379[6]
+```
+
+Schema for ALARM_STATS table is as below. When an alarm of particular severity is cleared,
+the corresponding severity counter is decremented. 
+```
+ALARM_STATS Table:
+==============================
+
+Key                       : id
+
+id                        : key {state}
+alarms                    : Number of active alarms {uint64}
+critical                  : Number of alarms of severity 'critical' {uint64}
+major                     : Number of alarms of severity 'major' {uint64}
+minor                     : Number of alarms of severity 'minor' {uint64}
+warning                   : Number of alarms of severity 'warning' {uint64}
+informational             : Number of alarms of severity 'informational' {uint64}
+
+127.0.0.1:6379[6]> hgetall "ALARM_STATS|state"
+ 1) "alarms"
+ 2) "1"
+ 3) "critical"
+ 4) "1"
+ 5) "major"
+ 6) "0"
+ 7) "minor"
+ 8) "0"
+ 9) "warning"
+10) "0"
+
+```
+### 3.1.8 Pull Model
+All NBIs - CLI, REST and gNMI - can pull contents of current alarm table and event history table.
+The following filters are supported:
+- ALL ( pulls all alarms)
+- Severity.
+- Recent alarms (eg., last 5 minutes, one hour, one day).
+- Records between two timestamps, one timestamp and end, and   beginning and a timestamp.
+- All records between two Sequence Numbers (incl end and begin)
+
+### 3.1.9 Supporting third party containers
+To support third party components ( e.g. FRR, teamd, DHCP Relay, LLDPd, ntpd etc ) which can not be modified to raise events, the following options are considered 
+and are being evaluated.
+1.  Patch the components
+    Create a patch for these components by adding libeventnotify library and invoke the API. This however, requires these patches need to be maintained in the code forever.
+
+2.  Listen to syslog messages
+    As many of these components raises syslog messages on an important event, a listener can be implemented to read incoming syslog messages and raise 
+    events based on the message.
+    This however is heavy on performance due to the fact that listener has to parse each syslog message. Also listener need to maintain a map of messages to 
+    event-id and need to be aware of resource and other specific details. It need to be aware of nuances of alarm raising/clearing if the component follows
+    any specific logic. 
+
+Approach 1 is preferred.
```

Comment:
I think HLD before it is approved should reflect the decision made. I don't see any update on that nor discussion with an agreement.
I don't think approach 1 is the right one as it as it will cause lots of maintenance as SONiC keep updating such components.
@lguohan  i appreciate if you can provide your input.

Reply:
Would appreciate a guidance on this one by the community.